code/modules/antagonists/_common/antag_team.dm

```diff
@@ -103,9 +104,6 @@ GLOBAL_LIST_EMPTY(antagonist_teams)
 		if(A.get_team() != src)
 			continue
 		return A
-	// If no matching antag datum was found, give them one.
-	if(antag_datum_type)
-		return member.add_antag_datum(antag_datum_type, src)
```

Comment:
This will break revs most likely, you should include another boolean parameter on get_antag_datum_from_member that determines if a new antag datum should be added (probably false by default, but it would be useful for add_member)

Reply:
I just moved this logic to `add_member`